datacosmos/errors.py

```diff
@@ -0,0 +1,19 @@
+"""Contains all DataCosmos exceptions."""
+
+
+class DataCosmosError(Exception):
```

Comment:
I understand it's generally considered best practice to use custom error types. Could you give a concrete example of how these types would be useful within our own code or in client code?

Reply:
I can't give a specific example, but it is just generally nice to give people the option of catching different types of errors and acting based on them. I also think it is a nice nudge for writers of the SDK to be mindful of the errors that are being raised and avoid just letting random library errors bubble up to the user without an explanation. `requests` is a serial offender for this as it doesn't include any information from the body of the response in its exceptions, so if a library author just uses `raise_for_status` then you get a really unhelpful error message that doesn't give any indication of what you should do to solve the problem